src/apps/explorer/components/common/Dropdown/Drodown.stories.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { ReactElement } from 'react'
+import styled, { css } from 'styled-components'
+
+import { Meta, Story } from '@storybook/react'
+import { GlobalStyles, ThemeToggler } from 'storybook/decorators'
+
+import { Dropdown, DropdownProps, DropdownOption, DropdownDirection } from './index'
+
+export default {
+  title: 'ExplorerApp/Dropdown',
+  component: Dropdown,
+  decorators: [GlobalStyles, ThemeToggler],
+} as Meta
+
+const onChangeOption = (selectedOption: number | string): void => {
+  console.info(`Selected option ${selectedOption}`)
+}
+
+const PaginationTextCSS = css`
+  color: ${({ theme }): string => theme.textPrimary1};
+  font-size: ${({ theme }): string => theme.fontSizeDefault};
+  font-weight: normal;
+  white-space: nowrap;
+`
+
+const PaginationDropdownButton = styled.button`
+  ${PaginationTextCSS}
+  cursor: pointer;
+  white-space: nowrap;
+  background: none;
+  border: none;
+
+  &.selected {
+    background-color: transparent;
+    cursor: not-allowed;
+    font-weight: 400;
+    opacity: 0.5;
+    pointer-events: none;
+  }
+`
+
+const dropdownOptions = (items: Array<string | number>): Array<ReactElement> =>
+  items.map((value) => (
+    <DropdownOption key={value} onClick={(): void => onChangeOption(value)}>
+      {value}
+    </DropdownOption>
+  ))
+
+const Template: Story<DropdownProps> = (args) => <Dropdown {...args} />
+
+const textOptions = ['All Orders', 'Canceled', 'Filled']
+const currentTextOption = textOptions[0]
+export const DefaultDropdown = Template.bind({})
+DefaultDropdown.args = {
+  dropdownDirection: DropdownDirection.downwards,
+  currentItem: textOptions.findIndex((option) => option === currentTextOption),
+  items: dropdownOptions(textOptions),
+  dropdownButtonContent: <PaginationDropdownButton>{currentTextOption} ▼</PaginationDropdownButton>,
+}
+
+const LimiterPaginationWrapper = styled(Dropdown)`
+  .dropdown-options {
+    min-width: 60px;
+  }
+`
+const TemplateFlat: Story<DropdownProps> = (args) => <LimiterPaginationWrapper {...args} />
+const itemsPerPage = [15, 30, 50, 100]
+const pageSize = itemsPerPage[1]
+export const PagerLimiter = TemplateFlat.bind({})
```

Comment:
The main difference between both stories that I can see is the use of `TemplateFlat` vs `Template`.
The first story is still not updated when the selection changes, while the second works fine.

Reply:
The difference was the styled component wrapper, but I've taken the opportunity to add a status to keep the option selected.